endpoint/src/main.rs

```diff
@@ -65,35 +76,73 @@ fn oauth2_create_token(
 }
 
 #[derive(Debug)]
-pub struct Host<'r>(Option<&'r str>);
+pub struct Host(Option<String>);
 
 #[rocket::async_trait]
-impl<'r> FromRequest<'r> for Host<'r> {
+impl<'r> FromRequest<'r> for Host {
     type Error = ();
 
     async fn from_request(
         request: &'r rocket::Request<'_>,
     ) -> rocket::request::Outcome<Self, Self::Error> {
-        rocket::request::Outcome::Success(Host(request.headers().get("Host").next()))
+        rocket::request::Outcome::Success(Host(
+            request.headers().get("Host").next().map(str::to_string),
+        ))
+    }
+}
+
+async fn read_config(file_path: &Option<String>) -> Result<HashMap<String, Tenant>, String> {
+    if file_path.is_none() {
+        if let Ok(false) = tokio::fs::try_exists("Tenants.json").await {
```

Comment:
Can we refactor `Tenants.json` to be a `const` instead?

Reply:
Likewise, I'd like to very much propose for an all-lowercase filename, i.e. `tenants.json`